### PR TITLE
Closes #3 Add format config feature

### LINKED ISSUE #3: Configurable result format
[13/11/2015 15:18:13] JP Slavinsky: make it configurable to say what the format of the babble is
[13/11/2015 15:18:15] JP Slavinsky: e.g.
[13/11/2015 15:18:19] JP Slavinsky: adjective noun
[13/11/2015 15:18:24] JP Slavinsky: adjective adjective noun
[13/11/2015 15:18:27] JP Slavinsky: adjective adjective

### PR DESCRIPTION
**Pull request by joemsak**

### CODE REVIEW

File: lib/babbler.rb [hunk 1]
```diff
@@ -12,13 +12,11 @@ def babble(seed = nil)
 
       babble_words = []
 
-      Babbler.config.num_adjectives.times do
-        babble_words.push(adjectives[prng.rand(adjectives.length)])
+      Babbler.config.format.gsub(' ', '').split('').each do |f|
+        babble_words.push(send("format_#{f}")[prng.rand(adjectives.length)])
```
Comment (jpslav) [inline lib/babbler.rb]:
``` ruby
"a b\t\nc".split(/\s+/)
```

There's probably some ask/tell thing going on here but `format_a` is vague to me.  I'd rather just have a case or hash to map "a" to :adjectives and leave the method as `adjectives`.

Comment (jpslav) [inline lib/babbler.rb]: Need error handling too so don't get crazy "NameError" if `format_z` doesn't exist.

``` ruby
babble_words = Babbler.config.format.downcase.split(/\s+/).collect do |f|
  # this case statement can be in a different method, or can be a hash and if result nil raise an error.
  word_method = case f
  when "a"
    :adjectives
  when "n"
    :nouns
  else
    raise # some error
  end
  send(word_method)[blah]
end
```

Comment (jpslav) [inline lib/babbler.rb]: the above also uses collect instead of creating array and pushing onto it; also means you don't need strip at bottom.

**Comment by joemsak** [comment]
@jpslav feedback resolved

File: lib/babbler.rb [hunk 2]
```diff
@@ -12,12 +12,10 @@ def babble(seed = nil)
 
       babble_words = []
 
-      Babbler.config.num_adjectives.times do
-        babble_words.push(adjectives[prng.rand(adjectives.length)])
+      supported_format.collect do |f|
+        babble_words.push(send("format_#{f}")[prng.rand(adjectives.length)])
```
Comment (jpslav) [inline lib/babbler.rb]: set `babbler_words` equal to the collection, and take out the push.  Make `format_X` private methods (they don't make sense to an end user as public methods.

File: lib/babbler.rb [hunk 3]
```diff
@@ -10,14 +10,10 @@ class << self
     def babble(seed = nil)
       prng = Random.new(seed || Random.new_seed)
 
-      babble_words = []
-
-      Babbler.config.num_adjectives.times do
-        babble_words.push(adjectives[prng.rand(adjectives.length)])
+      babble_words = supported_formats.collect do |f|
+        send("format_#{f}")[prng.rand(adjectives.length)]
```
Comment (jpslav) [inline lib/babbler.rb]: checking `adjectives.length` even tho it might be nouns.

**Pull request merged**